components/clm/src/biogeophys/SnowHydrologyMod.F90

```diff
@@ -1167,475 +1287,612 @@ subroutine DivideSnowLayers(bounds, num_snowc, filter_snowc, &
-                mbc_pho(c,2) = mbc_pho(c,1)
-                moc_phi(c,1) = moc_phi(c,1)/2._r8
-                moc_phi(c,2) = moc_phi(c,1)
-                moc_pho(c,1) = moc_pho(c,1)/2._r8
-                moc_pho(c,2) = moc_pho(c,1)
-                mdst1(c,1) = mdst1(c,1)/2._r8
-                mdst1(c,2) = mdst1(c,1)
-                mdst2(c,1) = mdst2(c,1)/2._r8
-                mdst2(c,2) = mdst2(c,1)
-                mdst3(c,1) = mdst3(c,1)/2._r8
-                mdst3(c,2) = mdst3(c,1)
-                mdst4(c,1) = mdst4(c,1)/2._r8
-                mdst4(c,2) = mdst4(c,1)
-                rds(c,2) = rds(c,1)
-
-             end if
-          end if
-
-          if (msno > 1) then
-             if (is_lake) then
-                offset = lsadz
-             else
-                offset = 0._r8
-             end if
-             if (dzsno(c,1) > 0.02_r8 + offset) then
-                if (is_lake) then
-                   drr = dzsno(c,1) - 0.02_r8 - lsadz
-                else
-                   drr = dzsno(c,1) - 0.02_r8
-                end if
-                propor = drr/dzsno(c,1)
-                zwice = propor*swice(c,1)
-                zwliq = propor*swliq(c,1)
-
-                zmbc_phi = propor*mbc_phi(c,1)
-                zmbc_pho = propor*mbc_pho(c,1)
-                zmoc_phi = propor*moc_phi(c,1)
-                zmoc_pho = propor*moc_pho(c,1)
-                zmdst1 = propor*mdst1(c,1)
-                zmdst2 = propor*mdst2(c,1)
-                zmdst3 = propor*mdst3(c,1)
-                zmdst4 = propor*mdst4(c,1)
-
-                if (is_lake) then
-                   propor = (0.02_r8+lsadz)/dzsno(c,1)
-                else
-                   propor = 0.02_r8/dzsno(c,1)
-                endif 
-
-                swice(c,1) = propor*swice(c,1)
-                swliq(c,1) = propor*swliq(c,1)
-
-                mbc_phi(c,1) = propor*mbc_phi(c,1)
-                mbc_pho(c,1) = propor*mbc_pho(c,1)
-                moc_phi(c,1) = propor*moc_phi(c,1)
-                moc_pho(c,1) = propor*moc_pho(c,1)
-                mdst1(c,1) = propor*mdst1(c,1)
-                mdst2(c,1) = propor*mdst2(c,1)
-                mdst3(c,1) = propor*mdst3(c,1)
-                mdst4(c,1) = propor*mdst4(c,1)
-
-                if (is_lake) then
-                   dzsno(c,1) = 0.02_r8 + lsadz
-                else
-                   dzsno(c,1) = 0.02_r8
-                end if
-
-                mbc_phi(c,2) = mbc_phi(c,2)+zmbc_phi  ! (combo)
-                mbc_pho(c,2) = mbc_pho(c,2)+zmbc_pho  ! (combo)
-                moc_phi(c,2) = moc_phi(c,2)+zmoc_phi  ! (combo)
-                moc_pho(c,2) = moc_pho(c,2)+zmoc_pho  ! (combo)
-                mdst1(c,2) = mdst1(c,2)+zmdst1  ! (combo)
-                mdst2(c,2) = mdst2(c,2)+zmdst2  ! (combo)
-                mdst3(c,2) = mdst3(c,2)+zmdst3  ! (combo)
-                mdst4(c,2) = mdst4(c,2)+zmdst4  ! (combo)
+       if (.not. use_extrasnowlayers) then
```

Comment:
Here the code is:

```
if (.not. use_extrasnowlayers) then
   ! 469 lines of code
else
  ! 133 lines of code
end if
```

Given the extremely large number of lines in the `if` condition, it is very hard to figure out where the `else` statement starts. Please introduce new subroutines that split the wor as

```
if (.not. use_extrasnowlayers) then
   call <approach-1>
else
   call <approach-2>
end if
```